File: backend/app/models/events.py
from ..extensions import db
from sqlalchemy import Enum
from datetime import datetime


class Event(db.Model):
    __tablename__ = "events"
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(255), nullable=False)
    description = db.Column(db.Text, nullable=False)
    location = db.Column(db.String(255), nullable=False)
    start_time = db.Column(db.DateTime, nullable=False)
    end_time = db.Column(db.DateTime, nullable=False)

    created_by = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
    creator = db.relationship("User", backref="events")

    status = db.Column(
        Enum("pending", "confirmed", "canceled", name="event_status"),
        default="pending",
        nullable=False,
    )

    image_url = db.Column(db.String(500), nullable=False)

    total_tickets = db.Column(db.Integer, nullable=False)
    tickets_remaining = db.Column(db.Integer, nullable=False)

    category_id = db.Column(db.Integer, db.ForeignKey("categories.id"), nullable=False)
    category = db.relationship("Category", backref="events")

    created_at = db.Column(db.DateTime, default=datetime.now(), nullable=False)
    updated_at = db.Column(
        db.DateTime, default=datetime.now(), onupdate=datetime.now(), nullable=False
    )

File: backend/internal/models/events.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type EventStatus string

const (
	EventPending   EventStatus = "pending"
	EventConfirmed EventStatus = "confirmed"
	EventCanceled  EventStatus = "canceled"
)

type Event struct {
	ID               uint           `gorm:"primaryKey" json:"id"`
	Title            string         `gorm:"not null" json:"title"`
	Description      string         `gorm:"type:text;not null" json:"description"`
	Location         string         `gorm:"not null" json:"location"`
	StartTime        time.Time      `gorm:"not null" json:"start_time"`
	EndTime          time.Time      `gorm:"not null" json:"end_time"`
	CreatedByID      uint           `gorm:"not null" json:"created_by"`
	Creator          *User          `gorm:"foreignKey:CreatedByID" json:"creator,omitempty"`
	Status           EventStatus    `gorm:"type:varchar(20);default:'pending';not null" json:"status"`
	ImageURL         string         `gorm:"size:500;not null" json:"image_url"`
	TotalTickets     int            `gorm:"not null" json:"total_tickets"`
	TicketsRemaining int            `gorm:"not null" json:"tickets_remaining"`
	CategoryID       uint           `gorm:"not null" json:"category_id"`
	Category         *Category      `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Price            float64        `gorm:"type:numeric(10,2);default:0" json:"price"`
	CreatedAt        time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt        time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt        gorm.DeletedAt `gorm:"index" json:"-"`
}
